lib/pbench/server/auth/auth.py

```diff
@@ -1,25 +1,80 @@
+from configparser import NoOptionError
 import datetime
 from http import HTTPStatus
 import os
-from typing import Optional
+from typing import List, Optional
 
 from flask import abort, request
 from flask_httpauth import HTTPTokenAuth
 import jwt
 
+from pbench.server import PbenchServerConfig
 from pbench.server.auth import OpenIDClient, OpenIDClientError
 from pbench.server.database.models.active_tokens import ActiveTokens
 from pbench.server.database.models.users import User
 
 
+class InternalUser:
+    """
+    Internal user class for storing user related fields fetch
+    from OIDC token decode.
+    """
+
+    def __init__(
+        self,
+        id: str,
+        username: str,
+        email: str,
+        first_name: str = None,
+        last_name: str = None,
+        roles: List[str] = None,
```

Comment:
We're now targeting Python 3.9 and up for server code, so you can use `list[str]`, which I think looks more natural.